stock_move_location/wizard/stock_move_location.py

```diff
@@ -30,19 +31,38 @@ class StockMoveLocationWizard(models.TransientModel):
         comodel_name="stock.picking",
     )
 
-    @api.onchange('origin_location_id', 'destination_location_id')
-    def _onchange_locations(self):
-        self._clear_lines()
+    @api.model
+    def default_get(self, fields):
+        res = super().default_get(fields)
+        if self.env.context.get('active_model', False) != 'stock.quant':
+            return res
+        # Load data directly from quants
+        quants = self.env['stock.quant'].browse(
+            self.env.context.get('active_ids', False))
+        res['stock_move_location_line_ids'] = [(0, 0, {
+            'product_id': quant.product_id.id,
+            'move_quantity': quant.quantity,
+            'max_quantity': quant.quantity,
+            'origin_location_id': quant.location_id.id,
+            'lot_id': quant.lot_id.id,
+            'product_uom_id': quant.product_uom_id.id,
+            'custom': False,
+        }) for quant in quants]
+        res['origin_location_id'] = quants[:1].location_id.id
```

Comment:
Prefer using 'first' method:
```suggestion
        from odoo.fields import first
        res['origin_location_id'] = first(quants).location_id.id
```

Reply:
Hi Denis, I didn't know this function, but... ¿Can you explain me in this case why is better use first(with prefetch) than directly records[:1]?
I have done a profile for two cases and I do not see a big difference, even records[:1] is faster.. :smile: 
You have to take into account that I already access to records so the data is in cache.

**Profiler with records[:1]**
calls     queries   ms
1         0         0.02          @profile
                                  @api.model
                                  def default_get(self, fields):
1         1         1.46              res = super().default_get(fields)
1         0         0.01              if self.env.context.get('active_model', False) != 'stock.quant':
                                          return res
                                      # Load data directly from quants
1         0         0.03              quants = self.env['stock.quant'].browse(
1         0         0.02                  self.env.context.get('active_ids', False))
1         0         0.0               res['stock_move_location_line_ids'] = [(0, 0, {
                                          'product_id': quant.product_id.id,
                                          'move_quantity': quant.quantity,
                                          'max_quantity': quant.quantity,
                                          'origin_location_id': quant.location_id.id,
                                          'lot_id': quant.lot_id.id,
                                          'product_uom_id': quant.product_uom_id.id,
                                          'custom': False,
1         6         70.74             }) for quant in quants]
**_1         0         0.04              res['origin_location_id'] = quants[:1].location_id.id_**
1         0         0.0               return res

Total:
1         7         72.32   

**Profiler with method firts**
calls     queries   ms
1         0         0.01          @profile
                                  @api.model
                                  def default_get(self, fields):
1         1         1.25              res = super().default_get(fields)
1         0         0.01              if self.env.context.get('active_model', False) != 'stock.quant':
                                          return res
                                      # Load data directly from quants
1         0         0.02              quants = self.env['stock.quant'].browse(
1         0         0.02                  self.env.context.get('active_ids', False))
1         0         0.0               res['stock_move_location_line_ids'] = [(0, 0, {
                                          'product_id': quant.product_id.id,
                                          'move_quantity': quant.quantity,
                                          'max_quantity': quant.quantity,
                                          'origin_location_id': quant.location_id.id,
                                          'lot_id': quant.lot_id.id,
                                          'product_uom_id': quant.product_uom_id.id,
                                          'custom': False,
1         6         70.37             }) for quant in quants]
**_1         0         0.05              res['origin_location_id'] = first(quants).location_id.id_**
1         0         0.0               return res

Total:
1         7         71.74

Reply:
I agree, but as Odoo provides the helper method, it's safer to use it than indexes (and more logic as indexes are for lists not really for recordsets).

Reply:
Ok, Thanks!! :smile:

Reply:
@rousseldenis not really, as that notation is Python standard, and recordsets adopt that standards (the same as set logic operations).